### PR TITLE
Use pytest-dcos plugin

### PR DESCRIPTION
**Pull request by mellenburg**
## High-level description
This PR replaces the hacky "api_session_fixture.py" module with an externally supported DC/OS testing plugin. This plugin provides basic API session setup and is pluggable so that specific arguments can be injected based on the local configuration file (as is currently the case)

## Corresponding DC/OS tickets (obligatory)

https://jira.mesosphere.com/browse/QUALITY-1883

## Checklist for all PRs

  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review): [Diff](https://github.com/dcos/dcos-test-utils/compare/47825add34311ce12bebe6b6ff0993b6bee8295c...9a9067501d12ce84d67ba04b6a5ab99ebc793b23)
  - [x] Test Results: [Open](https://teamcity.mesosphere.io/viewLog.html?buildId=1007502&buildTypeId=DcosIo_DcosTestUtils_ToxIntegrationTest) [Enterprise](https://teamcity.mesosphere.io/viewLog.html?buildId=1007501&buildTypeId=DcosIo_DcosTestUtils_ToxIntegrationTestEnterprise)
  - [x] Code Coverage (if available): [tox](https://teamcity.mesosphere.io/viewLog.html?buildId=1007503&buildTypeId=DcosIo_DcosTestUtils_ToxDcosTestUtils)

### CODE REVIEW

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot bump-ee

**Comment by mellenburg** [comment]
@mesosphere-mergebot override-status teamcity/dcos/test/aws/cloudformation/simple https://jira.mesosphere.com/browse/DCOS_OSS-2115

**Comment by orsenthil** [comment]
@mesosphere-mergebot override-status teamcity/dcos/test/aws/cloudformation/simple https://jira.mesosphere.com/browse/DCOS_OSS-2115

File: packages/dcos-integration-test/extra/conftest.py [hunk 1]
```diff
@@ -1,18 +1,37 @@
...
-logger.setup(os.getenv('TEST_LOG_LEVEL', 'INFO'))
+from test_helpers import expanded_config
+
 log = logging.getLogger(__name__)
 
+pytest_plugins = ['pytest-dcos']
+
+
+@pytest.fixture(scope='session')
+def dcos_api_session(dcos_api_session_factory):
```
Comment (orsenthil) [inline packages/dcos-integration-test/extra/conftest.py]: You can type specify this object `dcos_api_session_factory`: Type

File: packages/dcos-integration-test/extra/conftest.py [hunk 2]
```diff
@@ -1,18 +1,37 @@
...
     check_json,
     wait_for_diagnostics_job,
     wait_for_diagnostics_list
 )
-logger.setup(os.getenv('TEST_LOG_LEVEL', 'INFO'))
+from test_helpers import expanded_config
+
 log = logging.getLogger(__name__)
 
+pytest_plugins = ['pytest-dcos']
```
Comment (orsenthil) [inline packages/dcos-integration-test/extra/conftest.py]: I think, this is a secret sauce for pytest to register the plugin `pytest-dcos`. A comment above could help the reader.

Comment (margaret) [reply packages/dcos-integration-test/extra/conftest.py]: I think this is standard. It's at the top of the "how to use plugins" page on the pytest docs https://docs.pytest.org/en/latest/plugins.html#requiring-loading-plugins-in-a-test-module-or-conftest-file

**Pull request merged**